api/proto/config/vm.proto

```diff
@@ -61,4 +63,9 @@ message VmConfig {
   // that amount of available physical CPUs that is defined with the 'vcpus'
   // parameter defined above.
   bool pin_cpu = 20;
+  // Maximum amount of memory in kbytes allowed for VMM to occupy, aka
```

Comment:
If I understood correctly how it works, this name if unfortunately very very confusing. "Maximum amount of memory in kbytes allowed for VMM to occupy". But actually this value is the overhead, not the total amount of memory that can be occupied by the VMM according to the formula: memory + vmm_maxmem. So why not call something like vmm_memoverhead? vmm_mxmem looks like the real maximum amount of memory that can be occupied by the VM, i.e., memory + overhead.

Please, correct me if I'm missing something....

Reply:
"overhead" is word which makes people nervous. They want to "reduce the overhead". I want to avoid using this word, because there is a lot of confusion in the company on the "qemu overhead" term. This is the max amount of memory which we reserve for VMM (not VM) to function. The overall limit for VM instance is: VMM maxmemory + VM ram. That what I wanted to express in this comment.

Reply:
Ok, so you can still avoid overhead and be more clear, vmm_extmem (extended memory), vmm_addmem (additional memory), I still keep my point that vmm_maxmem is very confusing because it can easily be understood as the maximum memory allowed for a VMM instead of the maximum additional memory (i.e., the amount of memory allowed in addition to the amount already configured).

Reply:
> it can easily be understood as the maximum memory allowed for a VMM
this is exactly the case, we limit the memory for the app instance.

Reply:
Exactly, but vmm_maxmem *is not* the maximum memory allowed to a VM. Instead, is the maximum amount of additional memory that can be allocated to a VM (besides the amount of memory already allocated). I'm not being pick, this parameter has a very confusing name, I bet that this can led to more confusion in the future.

Reply:
VMM is not a VM. See the option: VmmMemoryLimitInMiB. Same thing.

Reply:
Indeed, I was confusing VMM with VM. Now it's clear. Thanks for the explanation.